lib/beaker-hostgenerator/hypervisor/hcloud.rb

```diff
@@ -0,0 +1,27 @@
+require 'beaker-hostgenerator/data'
+require 'beaker-hostgenerator/hypervisor'
+require 'deep_merge/rails_compat'
+
+module BeakerHostGenerator
+  module Hypervisor
+
+    class Hcloud < BeakerHostGenerator::Hypervisor::Interface
+      include BeakerHostGenerator::Data
+
+      def generate_node(node_info, base_config, bhg_version)
+        # split ostype into string and array
+        # ostype is ubuntu2204, debian10, rocky9
+        os, version = node_info['ostype'].split(/(\D+)/).reject!(&:empty?)
+        base_config['image'] = case os
+        when 'ubuntu'
+          "#{os}-#{version[0, 2]}.#{version[2,2]}"
+        when 'centos'
+          version.to_i == 7 ? "#{os}-#{version}" : "#{os}-stream-#{version}"
```

Comment:
@ekohl I reversed the logic and also added a test case for centos7/8 to verify it's working correctly.